ComponentKit/DataSources/Common/CKComponentAnnouncerBase.mm

```diff
@@ -9,86 +9,7 @@
  */
 
 #import "CKComponentAnnouncerBase.h"
-
-@interface CKComponentAnnouncerBase()
-  {
-  @public
-    // we make this public, so it can be accessed from CK::AnnouncerHelper
-    
-    // shared pointer (reference counted) of a vector listeners
-    // we need it to be a shared pointer, since we want to be
-    // able atomically grab it for enumeration, even if someone
-    // else is modifying it in a different thread at the same time
-    // We use a vector instead of a hash mainly for 2 reasons:
-    //  1) dealing with __weak id as a key in a hash is more complicated
-    //  2) we assume that the number of listeners is relatively small, and
-    //     add/remove is not a frequent event.
-    // n.b. using boost::shared_ptr might lead to faster code, since it's lockless
-    std::shared_ptr<const std::vector<__weak id>> _listenerVector;
-  }
-@end
+#import "CKComponentAnnouncerBaseInternal.h"
 
 @implementation CKComponentAnnouncerBase
```

Comment:
Without the C++ this header is left with no docs at all.  At least a small docblock to explain what this is might be helpful?